ouroboros-consensus/src/Ouroboros/Consensus/Storage/LedgerDB/OnDisk.hs

```diff
@@ -636,13 +644,147 @@ data LedgerBackingStoreValueHandle m l = LedgerBackingStoreValueHandle
+
+    -- remove all diff elements that are <= to the greatest given key
+    doDropLTE ::
+         Ord k
+      => ApplyMapKind KeysMK k v
+      -> ApplyMapKind DiffMK k v
+      -> ApplyMapKind DiffMK k v
+    doDropLTE (ApplyKeysMK (HD.UtxoKeys ks)) (ApplyDiffMK (HD.UtxoDiff ds)) =
+        ApplyDiffMK
+      $ HD.UtxoDiff
+      $ case Set.lookupMax ks of
+          Nothing -> ds
+          Just k  -> Map.filterWithKey (\dk _dv -> dk > k) ds
+
+    -- NOTE: this is counting the deletions wrt disk.
+    numDeletesDiffMK :: ApplyMapKind DiffMK k v -> Int
+    numDeletesDiffMK (ApplyDiffMK (HD.UtxoDiff m)) =
+      getSum $ foldMap (Sum . oneIfDel) m
+      where
+        oneIfDel (HD.UtxoEntryDiff _v diffstate) = case diffstate of
+          HD.UedsDel       -> 1
+          HD.UedsIns       -> 0
+          HD.UedsInsAndDel -> 0
+
+    -- INVARIANT: nrequested > 0
+    --
+    -- (1) if we reached the end of the store, then simply yield the given diff
+    --     applied to the given values
+    -- (2) otherwise, the readset must be non-empty, since 'rqCount' is positive
+    -- (3) remove the greatest read key
+    -- (4) remove all diff elements that are >= the greatest read key
+    -- (5) apply the remaining diff
+    -- (6) (the greatest read key will be the first fetched if the yield of this
+    --     result is next passed as 'rqPrev')
+    --
+    -- Note that if the in-memory changelog contains the greatest key, then
+    -- we'll return that in step (1) above, in which case the next passed
+    -- 'rqPrev' will contain it, which will cause 'doDropLTE' to result in an
+    -- empty diff, which will result in an entirely empty range query result,
+    -- which is the termination case.
+    --
+    -- TODO this @definitelyNoMoreToFetch@ logic leads to an extra roundtrip
+    -- when the changelog contains the greatest key. That'll be rare, and should
+    -- be in-expensive since the requisite pages would have been fetched by the
+    -- previous roundtrip (which was fruitful). But it seems sloppy. Switching
+    -- from 'rqPrev' to 'rqStart' would eliminate the extra rountrip and
+    -- simplify this code. However, it would also complicate the return type,
+    -- since the " next " range query wouldn't be defined by the keys of the "
+    -- previous " range query's results: we'd have to explicitly include the "
+    -- new next key " in the result of the range query. So maybe this should
+    -- stay as-is? Maybe some existing general wisdom out there about range
+    -- queries could guide us here.
+    doFixupReadResult ::
+         Ord k
+      => Int
+      -- ^ TODO: what is this?
+      -> ApplyMapKind DiffMK   k v
+      -- ^ TODO: what is this?
+      -> ApplyMapKind ValuesMK k v
+      -- ^ TODO: what is this?
+      -> ApplyMapKind ValuesMK k v
+    doFixupReadResult
+      nrequested
+      (ApplyDiffMK (HD.UtxoDiff ds))
+      (ApplyValuesMK (HD.UtxoValues vs)) = -- TODO: it's hard to see where the fundef begins
```

Comment:
I forgot to delete this. I wanted to ask where that equal sign should be. To me, it's hard to see where the function definition begins. Could this be an alternative? 

```haskell
    doFixupReadResult
      nrequested
      (ApplyDiffMK (HD.UtxoDiff ds))
      (ApplyValuesMK (HD.UtxoValues vs)) 
    = 
        let includingAllKeys              =
              HD.forwardValues (HD.UtxoValues vs) (HD.UtxoDiff ds)
            definitelyNoMoreToFetch       = Map.size vs < nrequested
            assertDefinitelyNoMoreToFetch =
              if Map.size vs < nrequested
              then id
              else error $ "Size of values " <> show (Map.size vs) <> ", nrequested " <> show nrequested
        in ...
```

Reply:
Certainly, or maybe have the arguments on one line and pattern match/extract `ds` and `vs` somewhere else?